fdbserver/TenantBalancer.actor.cpp

```
@@ -1049,21 +1049,35 @@ ACTOR Future<ReceiveTenantFromClusterReply> startSourceMovement(
-		                                                         self->connRecord->getConnectionString().toString()),
-		                         &TenantBalancerInterface::receiveTenantFromCluster),
-		                     record->onAbort()));
+		ReceiveTenantFromClusterReply r = wait(
+		    timeoutError(waitOrError(sendTenantBalancerRequest(record->getPeerDatabase(),
+		                                                       ReceiveTenantFromClusterRequest(
+		                                                           record->getMovementId(),
+		                                                           record->getSourcePrefix(),
+		                                                           record->getDestinationPrefix(),
+		                                                           self->connRecord->getConnectionString().toString()),
+		                                                       &TenantBalancerInterface::receiveTenantFromCluster),
+		                             record->onAbort()),
+		                 SERVER_KNOBS->TENANT_BALANCER_OPERATION_TIMEOUT));
 
 		reply = r;
 	} catch (Error& e) {
 		state Error destError = e;
 		if (cleanupOnError) {
-			wait(self->clearMovementRecord(mutableRecord));
+			try {
+				wait(timeoutError(self->clearMovementRecord(mutableRecord),
+				                  SERVER_KNOBS->TENANT_BALANCER_OPERATION_TIMEOUT));
+			} catch (Error& cleanupErr) {
+				TraceEvent(SevWarn, "TenantBalancerStartMovementCleanupFailed", self->tbi.id())
+				    .error(cleanupErr)
+				    .detail("MovementId", record->getMovementId())
+				    .detail("SourcePrefix", record->getSourcePrefix())
+				    .detail("DestinationPrefix", record->getDestinationPrefix())
+				    .detail("DestinationConnectionString",
+				            record->getPeerDatabase()->getConnectionRecord()->getConnectionString().toString());
+				record->setMovementError(format("Failed to cleanup movement: %s", cleanupErr.what()));
+				throw cleanupErr;
```

I think it may be better to throw the original error, or in other words don't do anything here and fall through.

---

No problem